sovrin_client/agent/agent.py

```diff
@@ -220,14 +220,21 @@ def createAgent(agentClass, name, wallet=None, basedirpath=None, port=None,
                          ha=("0.0.0.0", clientPort),
                          basedirpath=basedirpath)
 
-    return agentClass(basedirpath=basedirpath,
-                      client=client,
-                      wallet=wallet,
-                      port=port,
-                      loop=loop)
+    try:
+        return agentClass(basedirpath=basedirpath,
+                          client=client,
+                          wallet=wallet,
+                          port=port,
+                          loop=loop)
+    except AddressAlreadyInUse as e:
+        msg = getFormattedErrorMsg(str(e))
+        logger.error(msg)
```

Comment:
Shouldn't we be throwing the exception later, like when the agent is started? When the agent is started, let's let that be the time that the ip/port is allocated.